File: Java Spring Boot Project/CompanyDirectory/src/main/java/com/mindthecode/CompanyDirectory/models/responses/AllPositionsResponse.java
package com.mindthecode.CompanyDirectory.models.responses;

import java.util.List;

public class AllPositionsResponse {
    private List<PositionResponse> positions;

    public AllPositionsResponse(List<PositionResponse> positions) {
        this.positions = positions;
    }

    public List<PositionResponse> getPositions() {
        return positions;
    }

    public void setPositions(List<PositionResponse> positions) {
        this.positions = positions;
    }

    @Override
    public String toString() {
        return "AllPositionsResponse{" +
                "positions=" + positions +
                '}';
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/responses/all_positions_response.dart
import 'package:company_directory_app/models/responses/position_response.dart';

class AllPositionsResponse {
  final List<PositionResponse> positions;

  AllPositionsResponse({this.positions});

  factory AllPositionsResponse.fromJson(Map<String, dynamic> json) {
    var list = json['positions'] as List;
    List<PositionResponse> itemsList = list.map((i) => PositionResponse.fromJson(i)).toList();

    return AllPositionsResponse(positions: itemsList);
  }
}
